source/frontend/src/features/mapSideBar/acquisition/tabs/agreement/common/UpdateAcquisitionAgreementView.tsx

```diff
@@ -0,0 +1,118 @@
+import axios, { AxiosError } from 'axios';
+import { Formik } from 'formik';
+import styled from 'styled-components';
+
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import SidebarFooter from '@/features/mapSideBar/shared/SidebarFooter';
+import { getCancelModalProps, useModalContext } from '@/hooks/useModalContext';
+import { IApiError } from '@/interfaces/IApiError';
+import { ApiGen_Concepts_Agreement } from '@/models/api/generated/ApiGen_Concepts_Agreement';
+
+import AcquisitionAgreementForm from '../form/AcquisitionAgreementForm';
+import { AcquisitionAgreementFormYupSchema } from '../form/AcquisitionAgreementFormYupSchema';
+import { AcquisitionAgreementFormModel } from '../models/AcquisitionAgreementFormModel';
+
+export interface IUpdateAcquisitionAgreementViewProps {
+  isLoading: boolean;
+  initialValues: AcquisitionAgreementFormModel | null;
```

Comment:
kind of nit. The container does not necessarily know what the initial values should be given the form is the one that is responsible of putting the data in whatever way it needs.

Reply:
this comment makes no sense.

Reply:
What part of it lost you? I can rephrase it if necessary.
I summary, there could be a better separation of concerns in this code.

Reply:
The view is the same for both create and update containers ergo the view receives a set values (model) depending on the container that it's being invoked. The form (view) is just a representation.